stretch_moveit_config/demo.md

```diff
@@ -0,0 +1,95 @@
+
+## Demo with Stretch Robot
+
+### Installing ROS 2
+By default, Stretch RE1 ships with Ubuntu 18.04, ROS Melodic, and Python2 packages. These steps will install a second operating system with Ubuntu 20.04, ROS Noetic, ROS 2 Galactic (where MoveIt 2 is supported), and Python3 packages.
+
+1. A new firmware called protocol 1 (p1) for Stretch adds support for spline trajectories, a feature used by MoveIt2. Use the [firmware update guide](https://github.com/hello-robot/stretch_firmware/blob/master/tutorials/docs/updating_firmware.md) to install the new firmware.
+
+2. Use the [Noetic installation instructions](https://github.com/hello-robot/stretch_ros/blob/dev/noetic/install_noetic.md) to create the second Ubuntu 20.04 partition, and create a ROS2 Galactic workspace with MoveIt 2 installed.
+
+3. Boot into the new Ubuntu 20.04 partition, and edit the last few lines of the ~/.bashrc file to comment out sourcing of Noetic workspaces. When both versions of ROS are sourced, package conflicts arise. The result should look like:
+
+```
+#source /opt/ros/noetic/setup.bash
+source /opt/ros/galactic/setup.bash
+#source ~/catkin_ws/devel/setup.bash
+source ~/ament_ws/install/setup.bash
+```
+
+### Installing MoveIt 2 and Its Dependencies
+
+4. Clone the [stretch_ros2 repository](https://github.com/hello-robot/stretch_ros2) in your workspace and switch to the “feature/hybrid_planning” branch within stretch_ros2 using the following command.
+
+```
+cd ~/ament_ws/src
+git clone https://github.com/hello-robot/stretch_ros2.git
+git checkout feature/hybrid_planning
+```
+
+4. a. In the absence of support for stretch_calibration in ROS 2 (porting in progress which will make this step redundant), we need to manually move the calibrated urdf file named stretch.urdf from the stretch_description/urdf directory in stretch_ros (in ROS Melodic) to its counterpart in stretch_ros2 (in ROS 2 Galactic).
+
+4. b. We also need to move the controller_calibration_head.yaml file from the stretch_core/config directory in stretch_ros to its counterpart in stretch_ros2.
+
+5. In a new terminal, collect all of the packages to build from source. Use the command below to automate collection of these packages. Download the stretch_moveit2.repos file or copy and paste the contents of the file from [here](https://github.com/hello-robot/stretch_ros2/blob/feature/hybrid_planning/stretch_moveit_config/stretch_moveit2.repos).
+
+```
+cd ~/ament_ws/src
+vcs import < stretch_moveit2.repos
+```
+
+6. Download additional dependencies and build the workspace using these commands. This should install MoveIt 2 and everything else you need to run the tutorials!
+
+```
+cd ~/ament_ws
+rosdep install --from-paths src --ignore-src -r --rosdistro galactic -y
+cd ~/ament_ws
+colcon build
+source ~/ament_ws/install/setup.bash
+```
```

Comment:
A robot upgrade will automatically handle these steps. If the robot has an outdated 20.04, let's recommend following the [Adding a ROS Workspace](https://docs.hello-robot.com/0.2/stretch-install/docs/ros_workspace/#ubuntu-2004) guide to create a new ament_ws.